### PR TITLE
MPC: bump

### PR DESCRIPTION
**Pull request by extrowerk**

### CODE REVIEW

File: dev-libs/mpc/mpc-1.1.0.recipe [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+extends the principles of the IEEE-754 standard for fixed precision real \
+floating point numbers to complex numbers, providing well-defined semantics \
+for every operation. At the same time, speed of operation at high precision \
+is a major design goal."
+HOMEPAGE="http://www.multiprecision.org"
+COPYRIGHT="2006-2017 Andreas Enge, Mickaël Gastineau, Philippe Théveny and \
+Paul Zimmermann"
+LICENSE="GNU LGPL v3"
+REVISION="1"
+SOURCE_URI="https://ftp.gnu.org/gnu/mpc/mpc-$portVersion.tar.gz"
```
Comment (fbrosson) [inline dev-libs/mpc/mpc-1.1.0.recipe]:
```diff
-SOURCE_URI="https://ftp.gnu.org/gnu/mpc/mpc-$portVersion.tar.gz"
+SOURCE_URI="https://ftpmirror.gnu.org/mpc/mpc-$portVersion.tar.gz"
```

File: dev-libs/mpc/mpc-1.1.0.recipe [hunk 2]
```diff
@@ -0,0 +1,71 @@
+SUMMARY="Arithmetics of complex numbers with arbitrarily high precision"
+DESCRIPTION="GNU Mpc is a C library for the arithmetic of complex numbers \
+with arbitrarily high precision and correct rounding of the result. It \
+extends the principles of the IEEE-754 standard for fixed precision real \
+floating point numbers to complex numbers, providing well-defined semantics \
+for every operation. At the same time, speed of operation at high precision \
+is a major design goal."
+HOMEPAGE="http://www.multiprecision.org"
```
Comment (fbrosson) [inline dev-libs/mpc/mpc-1.1.0.recipe]:
```diff
-HOMEPAGE="http://www.multiprecision.org"
+HOMEPAGE="http://www.multiprecision.org/mpc/"
```
(We get a `HTTP/1.1 301 Moved Permanently` with `curl -I`.)

File: dev-libs/mpc/mpc-1.1.0.recipe [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+}
+
+defineDebugInfoPackage mpc$secondaryArchSuffix \
+	$libDir/libmpc.so.$libVersion
+
+INSTALL()
+{
+	make install
+	rm $libDir/libmpc.la
+	prepareInstalledDevelLibs libmpc
```
Comment (fbrosson) [inline dev-libs/mpc/mpc-1.1.0.recipe]:
```diff
-	prepareInstalledDevelLibs libmpc
+	prepareInstalledDevelLib libmpc
```

File: dev-libs/mpc/mpc-1.1.0.recipe [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+LICENSE="GNU LGPL v3"
+REVISION="1"
+SOURCE_URI="https://ftp.gnu.org/gnu/mpc/mpc-$portVersion.tar.gz"
+CHECKSUM_SHA256="6985c538143c1208dcb1ac42cedad6ff52e267b47e5f970183a3e75125b43c2e"
+
+ARCHITECTURES="x86_gcc2 x86 x86_64"
+SECONDARY_ARCHITECTURES="x86"
+
+libVersion=3.1.0
+libVersionCompat="$libVersion compat >= 1"
```
Comment (alaviss) [inline dev-libs/mpc/mpc-1.1.0.recipe]:
```diff
-libVersionCompat="$libVersion compat >= 1"
+libVersionCompat="$libVersion compat >= ${libVersion%%.*}"

File: dev-libs/mpc/mpc-1.1.0.recipe [hunk 5]
```diff
@@ -0,0 +1,71 @@
+SUMMARY="Arithmetics of complex numbers with arbitrarily high precision"
+DESCRIPTION="GNU Mpc is a C library for the arithmetic of complex numbers \
+with arbitrarily high precision and correct rounding of the result. It \
+extends the principles of the IEEE-754 standard for fixed precision real \
+floating point numbers to complex numbers, providing well-defined semantics \
+for every operation. At the same time, speed of operation at high precision \
+is a major design goal."
+HOMEPAGE="http://www.multiprecision.org"
+COPYRIGHT="2006-2017 Andreas Enge, Mickaël Gastineau, Philippe Théveny and \
```
Comment (alaviss) [inline dev-libs/mpc/mpc-1.1.0.recipe]: A small grep on the source show that copyright is instead attributed to `INRIA`

```diff
-COPYRIGHT="2006-2017 Andreas Enge, Mickaël Gastineau, Philippe Théveny and \
-Paul Zimmermann"
+COPYRIGHT="2002-2011 INRIA"

File: dev-libs/mpc/mpc-1.1.0.recipe (hunk 4 above)
Comment (fbrosson) [reply dev-libs/mpc/mpc-1.1.0.recipe]: I think there is some historical reason for this non-standard compat.
(@korli is Haiku's main gcc specialist and I think he can explain why we need to keep the `>= 1`.)

Comment (alaviss) [reply dev-libs/mpc/mpc-1.1.0.recipe]: It's just a typo @fbrosson 

https://github.com/haikuports/haikuports/blob/7cfe089a5f33297b9d3e361227f60ab12775ee0a/dev-libs/mpc/mpc-1.0.3.recipe#L21

File: dev-libs/mpc/mpc-1.1.0.recipe [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+{
+	make install
+	rm $libDir/libmpc.la
+	prepareInstalledDevelLib libmpc
+	packageEntries devel $developDir
+}
+
+TEST()
+{
+	make $jobArgs check
```
Comment (fbrosson) [inline dev-libs/mpc/mpc-1.1.0.recipe]: Shall we drop the `documentation/info/mpc.info` on secondary architectures and disable parallel builds in `TEST()`?
```diff
 INSTALL()
 {
 	make install
 	rm $libDir/libmpc.la
+	if [ -n "$secondaryArchSuffix" ]; then
+		rm -rf $infoDir
+	fi
 	prepareInstalledDevelLib libmpc
 	packageEntries devel $developDir
 }
 
 TEST()
 {
-	make $jobArgs check
+	make check
 }
```

Everything else looks OK. Built on x86_gcc2, x86 second arch and x86_64.
Tested on x86 secondary arch and x86_64. All 67 tests pass on these architectures.

On x86_gcc2, however, gcc2 cannot build `make check`.

Comment (fbrosson) [reply dev-libs/mpc/mpc-1.1.0.recipe]: Oops, I just notice this line:
```
/boot/system/develop/headers/posix/signal.h:75: warning: ANSI C doesn't support unnamed structs/unions
```
when I run `hp --test mpc` on x86_gcc2:
```
In file included from /boot/system/develop/headers/bsd/signal.h:9,
                 from /boot/system/develop/headers/posix/sys/wait.h:10,
                 from /boot/system/develop/headers/bsd/sys/wait.h:9,
                 from /boot/system/develop/headers/posix/stdlib.h:14,
                 from /boot/system/develop/headers/bsd/stdlib.h:9,
                 from /sources/mpc-1.1.0/tests/random.c:26:
/boot/system/develop/headers/posix/signal.h:75: warning: ANSI C doesn't support unnamed structs/unions
/sources/mpc-1.1.0/tests/random.c: In function `test_random_d':
/sources/mpc-1.1.0/tests/random.c:234: initializer element is not computable at load time
/sources/mpc-1.1.0/tests/random.c:234: (near initialization for `mpfr_x[0]')
make[2]: *** [Makefile:1303: random.lo] Error 1
make[2]: Leaving directory '/sources/mpc-1.1.0/tests'
```
Shouldn't the [headers/posix/signal.h](https://github.com/haiku/haiku/blob/master/headers/posix/signal.h#L75) shipped in x86_gcc2 be gcc2-compatible?

File: dev-libs/mpc/mpc-1.1.0.recipe [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+	"
+
+BUILD()
+{
+	runConfigure ./configure
+	make $jobArgs
+}
+
+defineDebugInfoPackage mpc$secondaryArchSuffix \
+	$libDir/libmpc.so.$libVersion
```
Comment (alaviss) [inline dev-libs/mpc/mpc-1.1.0.recipe]: This should be done before `BUILD()`.

File: dev-libs/mpc/mpc-1.1.0.recipe (hunk 6 above)
Comment (alaviss) [reply dev-libs/mpc/mpc-1.1.0.recipe]: @fbrosson It's compatible in the sense that gcc2 doesn't panic upon meeting. I think the warning could be easily silenced. You should file a ticket on Trac for this.

**Pull request merged**